### PR TITLE
[WEB][API] Correct sortType and order for host in realtime

### PR DESCRIPTION
**Pull request by MelissaB79**
… realtime host

## Description

To make a sort type on host status with the API

**Fixes** # (MON-5668)

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [X] 19.10.x
- [X] 20.04.x
- [X] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

With this modification, it is now possible to sort the realtime host by one field. Before, the sortType parameter could take only the values "desc" and "asc".

### CODE REVIEW

File: www/api/class/centreon_realtime_hosts.class.php [hunk 1]
```diff
@@ -141,20 +141,20 @@ protected function setHostFilters()
...
+            if (strtolower($this->arguments['order']) === 'asc' ||
+                strtolower($this->arguments['order']) === 'desc') {
+                $this->order = $this->arguments['order'];
             } else {
-                throw new \RestBadRequestException('Bad sort type parameter');
+                throw new \RestBadRequestException('Bad order parameter');
             }
         } else {
-            $this->sortType = null;
+            $this->order = null;
```
Comment (sc979) [inline www/api/class/centreon_realtime_hosts.class.php:152]:
```suggestion
            $this->order = 'asc';
```

Comment (sc979) [reply www/api/class/centreon_realtime_hosts.class.php:152]: I think, that a default value should be set if none given

**Comment by sc979** [comment]
Hi @MelissaB79 ,

Thanks for the contribution :100:

Your contribution will be discussed during the next weekly sessions of refinement with the team.

Thanks :+1:

Regards,

File: www/api/class/centreon_realtime_hosts.class.php (hunk 1 above)
Comment (jeremyjaouen) [reply www/api/class/centreon_realtime_hosts.class.php:152]: @sc979  a default value is already given later in the file:
```
            $q = 'ASC';
            if (isset($this->order) && strtoupper($this->order) === 'DESC') {
                $q = 'DESC';
            }
```

**Comment by jeremyjaouen** [comment]
Hello @MelissaB79, Your PR has been merged with this one: https://github.com/centreon/centreon/pull/8837 , we've also add some missing fields into the sortType.

Thanks for contributing 👍 

Regards,

**Pull request closed without merging**